src/protocol/payload/addr.rs

```diff
@@ -82,8 +84,8 @@ impl NetworkAddr {
     }
 
     pub(super) fn decode(bytes: &mut Cursor<&[u8]>) -> io::Result<Self> {
-        let timestamp = i64::from_le_bytes(read_n_bytes(bytes)?);
-        let dt = DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(timestamp, 0), Utc);
+        let timestamp = u32::from_le_bytes(read_n_bytes(bytes)?);
+        let dt = DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(timestamp.into(), 0), Utc);
```

Comment:
I think we should drop the `DateTime` here completely, it's an unnecessary conversion we should only implement for debugging if we need a more human friendly format.

Reply:
We do still need to initialize new addresses with a timestamp value for tests. Which means we will likely use `chrono::Utc::now()` to get a reasonable value (which we have to convert anyway).

I also prefer the field indicating what it is - and the encoding / decoding already encapsulates what to do with the value.